flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/type/ArrayType.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.type;
+
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Type for Array.
+ */
+public class ArrayType implements InternalType {
+
+	private final InternalType elementType;
+	private final boolean primitive;
+
+	public ArrayType(InternalType elementType) {
+		this(elementType, false);
```

Comment:
Can we simply use `false` here? What if user passed in a LongType?

Reply:
BTW, why we need a primitive flag here?

Reply:
We need to distinguish `long[]` and `Long[]` which are different types. For example, if users have a udf with `eval(long[] arg)` signature, we have to give the right type `long[]` not `Long[]`.

Reply:
I think we can remove primitive flag, because we use BinaryArray internal, it is nullable, can cover all cases.